app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -783,10 +806,10 @@ public boolean matches(UserKeyRecord ukr) {
         //TODO: We can just delete the db entirely.
 
         Editor sharedPreferencesEditor = CommCareApplication._().getCurrentApp().getAppPreferences().edit();
+        sharedPreferencesEditor.putString(CommCarePreferences.LAST_LOGGED_IN_USER, null).commit();
 
-        sharedPreferencesEditor.putString(CommCarePreferences.LAST_LOGGED_IN_USER, null);
-
-        sharedPreferencesEditor.commit();
+        // manually clear file-backed fixture storage to ensure files are removed
+        CommCareApplication._().getFileBackedUserStorage("fixture", FormInstance.class).removeAll();

```

Comment:
Making sure to remove files for file-backed databases when a "clear user data" is performed